### PR TITLE
Add more operations & refactor

### PR DESCRIPTION
**Pull request by jbtrystram**
Added an API client to manage API tokens
Added an API client to handle application administration operations

Moved common methods into a shared trait between clients in the `util::Client` trait

### CODE REVIEW

File: src/registry/v1/client.rs [hunk 1]
```diff
@@ -1,17 +1,15 @@
...
-use serde::de::DeserializeOwned;
 use std::fmt::Debug;
 use tracing::instrument;
 use url::Url;
 
-/// A device registry client, backed by reqwest.
+/// A device registry client
 #[derive(Clone, Debug)]
-pub struct Client<TP>
+pub struct RegistryClient<TP>
```
Comment (ctron) [inline src/registry/v1/client.rs:12]: I would like to keep the name as just `Client`. True, `RegistryClient` would work as well, however now that mean a lot of renaming.

From the user's perspective: when only using registry stuff, one could simply use `Client`. When adding more clients, one can still use `registry::v1::Client` or `use registry::v1::Client as RegistryClient`.

So the `Registry` prefix felt a bit redundant, as in the `registry::v1` module, I would only expect "registry" clients anyway.

Comment (jbtrystram) [reply src/registry/v1/client.rs:12]: Good point

**Review by ctron — APPROVED** [review]
LGTM!

**Review by lulf — APPROVED** [review]
LGTM!

**Pull request merged**